pandas/io/parquet.py

```diff
@@ -351,11 +351,11 @@ def read(
 
         parquet_file = self.api.ParquetFile(path, **parquet_kwargs)
 
-        result = parquet_file.to_pandas(columns=columns, **kwargs)
-
-        if handles is not None:
-            handles.close()
-        return result
+        try:
+            return parquet_file.to_pandas(columns=columns, **kwargs)
```

Comment:
Might as well move `parquet_file = self.api.ParquetFile(path, **parquet_kwargs)` into the try block.

Reply:
Good point, done

Reply:
Can you please also add a test. Probably something similar to this test https://github.com/pandas-dev/pandas/blob/3cb70a5f0d1dc386f7806fca75abff6f098b1539/pandas/tests/io/parser/test_unsupported.py#L185

Reply:
Does `ensure_clean` ensure that there’s no file handle to that path?

On Linux, deleting a file doesn’t mean its file handles are closed …

So basically: I’m not sure how to test it.

Reply:
`ensure_clean` will try to unlink the file on `__exit__`. This will not detect open files on linux but on windows it will fail. Not the best test but it seems to work :) let me know if you have a better idea!